src/coreComponents/mesh/generators/WellBlock.cpp

```diff
@@ -25,474 +25,38 @@
 
 namespace geosx
 {
-WellBlock::WellBlock( const InternalWellGenerator & internalWellGenerator ) :
-  m_numElemsPerSegment( internalWellGenerator.getNumElementsPerSegment() ),
-  m_numNodesPerElem( internalWellGenerator.getNumNodesPerElement() ),
-  m_nDims( internalWellGenerator.getPhysicalDimensionsNumber() )
-{
-  // count the number of well elements to create
-  m_numElems = m_numElemsPerSegment * internalWellGenerator.getSegmentToPolyNodeMap().size( 0 );
-  m_numNodes = m_numElems + 1;
-
-  // resize the well element, node, and perforation arrays
-  m_elemCenterCoords.resize( m_numElems, 3 );
-  m_nextElemId.resize( m_numElems );
-  m_prevElemId.resize( m_numElems );
-  m_elemToNodesMap.resizeDimension< 0 >( m_numElems );
-  m_elemToNodesMap.resizeDimension< 1 >( m_numNodesPerElem );
-  m_elemVolume.resize( m_numElems );
 
-  m_nodeDistFromHead.resize( m_numNodes );
-  m_nodeCoords.resize( m_numNodes, 3 );
-
-  m_numPerforations = internalWellGenerator.getNumPerforations();
-  m_perfCoords.resize( m_numPerforations, 3 );
-  m_perfDistFromHead.resize( m_numPerforations );
-  m_perfTransmissibility.resize( m_numPerforations );
-  m_perfElemId.resize( m_numPerforations );
+WellBlock::WellBlock( string const & name, Group * const parent ):
+  WellBlockABC( name, parent )
+{}
 
+void WellBlock::importFieldsFromInternalWellGenerator( InternalWellGenerator const & internalWellGenerator ) {
```

Comment:
Eventually, while heavy, sorry, we'll have to replace this with setters: we do not want `WellBlock` to depend on `InternalWellGenerator`, there will be other sources of information soon (or not so soon, but there will be 🤣 )!

Your IDE will surely let you define those setters automatically.

Reply:
Emacs macros can also help :p; Done